### ISSUE TITLE
tag duplicate in a same post & `#` in code block is not tag

### ISSUE THREAD

**Issue by qaqland** [toddsundsted/ktistec#97]
my code block has some c macro

![image](https://github.com/toddsundsted/ktistec/assets/62464571/72bf2693-35a5-4272-b3c2-7e3e75f00594)

https://qaq.land/tags/pragma

Ktistec Version 2.0.0-10

**Comment by toddsundsted** [comment]
good catch! thanks!

**Comment by toddsundsted** [comment]
fixed in a6ea1385. can you confirm?

**Comment by qaqland** [comment]
new post fixed, what about old post🤔

**Comment by qaqland** [comment]
could you access my instance https://qaq.land ?

it still has https://qaq.land/tags/define and https://qaq.land/tags/pragma

```
2024-04-30 17:24:26 UTC 200 GET /tags/define 3.72ms
2024-04-30 17:24:29 UTC 200 GET /tags/pragma 5.56ms
2024-04-30 17:26:02 UTC 409 POST /actors/qaq/inbox 510.07ms
2024-04-30 17:27:53 UTC 404 GET /tags/pragma2 3.72ms
```

**Comment by toddsundsted** [comment]
interesting. that should have deleted the tags in addition to just the tag markup. i'll take a look at it

**Comment by toddsundsted** [comment]
yep, found a bug. when _all_ tags are removed from the post none of the tags get removed.

**Comment by toddsundsted** [comment]
ok, this bug is fixed in a824ce02. if you edit those posts and then update, it should correctly remove the hashtag database entries, as well.

**Comment by toddsundsted** [comment]
@qaqland these two fixes should address the issue you mentioned. can you confirm?

**Comment by qaqland** [comment]
yes, works well. Thank you!
